### PR TITLE
A Worker class for loop routine jobs

### PR DESCRIPTION
**Pull request by Billy4195**
The Worker class can be used to execute the routine jobs.
The routine jobs in arrange_schedule could be replaced by Worker class, since the class can reduce the redundant code for forking child process, checking fork result.

### CODE REVIEW

**Comment by Billy4195** [comment]
@icecat2012 
1. I change the return value's name, it should be called as `fork_failed` 
2. I catch the exceptions in the `job` function which is ***expired_data_check()*** now, and then it won't cause infinite loop.

**Comment by icecat2012** [comment]
Well, I suggest use 
if pid == 0:
  exit(0)
else:
  return 1
after "except Exception as e:"
It's good to see try catch in self.fn().
However, I think we had better code second line of defense, which guarantee future programmer's mistake wont cause infinite fork.

**Comment by Billy4195** [comment]
Yes, your suggestion is better, I will change it.
But I can't understand the meaning of `second line of defense`

**Comment by icecat2012** [comment]
work.py line 19 "if pid and pid == 0:" seems incorrect? This condition will never be true.
I think it should be "if pid is not None and pid == 0:"

File: arrange_schedule.py [hunk 1]
```diff
@@ -1347,28 +1348,12 @@ def set_system_log(json_obj):
...
-        else: #Parent
-            alarm_expire_data_check = raw_time + 1800.0
-    except:
-        receive_obj["result"] = "fail"
-        receive_obj["error"] = "fork1 error"
+    receive_obj = expire_data_check_()
+    if receive_obj["result"] == "success":
+        "DO NOTHING"
+    else :
+        receive_obj["error"] = "expire_data_check : " + receive_obj["error"]
```
Comment (stvreumi) [inline arrange_schedule.py]: If all the string representation should using `.format()` method, maybe line 1355 could change to : 
`receive_obj["error"] = "expire_data_check : {message}".format(message=receive_obj["error"])`

Comment (Billy4195) [inline arrange_schedule.py]: Ok, I will modify it.

**Comment by icecat2012** [comment]
@chenyang14 

I think this is an important commit. 
If it works well, future fork functions will consider it as a good example.
Please review and check it out whether it need to be improved or not.

**Pull request merged**